WELD-2386 allow to define bean discovery mode for synthetic archives …

manovotn opened the pull request:
…in SE environment.

mkouba commented on environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java:
```diff
@@ -662,6 +664,17 @@ public Weld addServices(Service... services) {
+     *
+     * @return  self
+     */
+    public Weld setSyntheticArchiveBeanDiscoveryMode(String mode) {
```
I think `setBeanDiscoveryMode()` should be ok. We don't use `syntheticArchive` in other methods neither.

manovotn replied:
Hmm, can do I just wanted to make it clear that this makes sense only once you `disableDiscovery()`

mkouba replied:
Also `BeanDiscoveryMode` is part of SPI. So we could probably use the enum as the param.

tremes replied:
+1 to enum

tremes commented:
It's just minor, but I would add new SE tests to `environments/se/tests`

manovotn commented:
@tremes I don' understand, that's where the tests are, isn't it?

tremes commented:
Ah sorry you're right. I thought it was in `environments/se/core/test/`.

manovotn commented:
Since weekend are long...a note to myself:
* Martin suggested a filter instead of mode as parameter - we could expand on this.
* I removed all notions of mode `NONE` as an option because it makes no sense in SE.
** Therefore I added an exception/test for IAE when you use this mode

mkouba commented on environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java:
```diff
@@ -662,6 +665,21 @@ public Weld addServices(Service... services) {
 
     /**
+     * Sets the bean discovery mode for synthetic bean archive. Default mode is ALL.
+     * @param mode Bean discovery mode as string, possible values are ALL, ANNOTATED
```
It's not a string anymore...

mkouba commented on environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java:
```diff
@@ -662,6 +665,21 @@ public Weld addServices(Service... services) {
+     * @param mode Bean discovery mode as string, possible values are ALL, ANNOTATED
+     *
+     * @return  self
+     */
```
Add `@throws IllegalArgumentException If...`

mkouba commented:
The idea of a filter is probably an overkill.
